pulsar-common/src/main/java/org/apache/pulsar/common/util/collections/ConcurrentTripleLong2LongHashMap.java

```diff
@@ -0,0 +1,94 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pulsar.common.util.collections;
+
+import java.util.HashMap;
+
+public class ConcurrentTripleLong2LongHashMap {
+    public class TripleLong{
+        public long first;
+        public long second;
+        public long third;
+        @Override
+        public int hashCode() {
+            return Long.hashCode(first) ^ Long.hashCode(second) ^ Long.hashCode(third);
+        }
+        @Override
+        public boolean equals(Object obj) {
+            if(obj instanceof TripleLong){
+                TripleLong other = (TripleLong) obj;
+                return first == other.first && second == other.second && third == other.third;
+            }
+            return false;
+        }
+    }
+
+    private HashMap<TripleLong, Long> map;
+    public ConcurrentTripleLong2LongHashMap(){
```

Comment:
Why does the name have the "Concurrent" prefix if this isn't supporting concurrent access?

Reply:
I use `HashMap` for now, which is not the final version.
I think we can implement a simillar concurrent map class like `ConcurrentLongLongPairHashMap`, each entry take up 4 long space and without any wrapper.